runtimes/core/src/api/auth/mod.rs

```diff
@@ -161,6 +164,30 @@ impl Authenticator {
             }
         };
 
+        // Cookies.
+        if let Some(schema) = &self.schema.cookie {
+            let mut inbound_cookies = cookie::CookieJar::new();
+            for raw in inbound.headers().get_all(http::header::COOKIE) {
+                if let Ok(raw) = String::from_utf8(raw.as_bytes().to_vec()) {
```

Comment:
Is `raw` not a string already? What  type is it? A bit strange to need to first convert to bytes and then to string

Reply:
it is a HeaderValue which is a bit annoying to deal with, we can use `to_str` and then unwrap_or_default. I can see if I can make it a bit less code

Reply:
Mainly trying to avoid unnecessary allocations; I guess if `to_str` fails we wouldn't be able to parse it as a string using the current code either

Reply:
ah wait, I thought I had used from_utf8_lossy here to avoid errors, but I see that I havent. I'll have a look and see if I can clean that up a bit